scipy/stats/_distribution_infrastructure.py

```diff
@@ -3717,15 +3739,26 @@ def _overrides(method_name):
 
 def _make_distribution_custom(dist):
     parameters = []
-    support = getattr(dist, 'support')
 
     for name, info in dist.parameters.items():
         domain = _RealDomain(endpoints=info['endpoints'],
                              inclusive=info['inclusive'])
         param = _RealParameter(name, domain=domain)
         parameters.append(param)
 
-    _x_support = _RealDomain(endpoints=support, inclusive=(True, True))
+    if callable(dist.support):
+        def support(**parameter_values):
+            a, b = dist.support(**parameter_values)
+            return np.asarray(a)[()], np.asarray(b)[()]
+    else:
+        support = dist.support
+
+    if hasattr(dist, "support_inclusive"):
+        inclusive = dist.support_inclusive
```

Comment:
Do we actually want to allow letting users decide if the support is `inclusive`?. If so, is `support_inclusive` the best attribute name to use here? I added this because the `cdf` and `pdf` formulas for the generalized extreme value distribution used in the tests isn't defined on a finite endpoint of the support in some cases.

Reply:
OK, yeah. I thought of that too but was worried it might be too cumbersome. I'm happy with that.